core/src/main/scala/org/apache/spark/status/protobuf/JobDataWrapperSerializer.scala

```diff
@@ -79,16 +80,16 @@ class JobDataWrapperSerializer extends ProtobufSerDe[JobDataWrapper] {
   }
 
   private def deserializeJobData(info: StoreTypes.JobData): JobData = {
-    val description = getOptional(info.hasDescription, info.getDescription)
+    val description = getOptional(info.hasDescription, () => weakIntern(info.getDescription))
```

Comment:
I wonder if it's worthwhile to intern strings that aren't likely to be repeated. Descriptions don't seem to be that case. IDs and names also seem more unique-ish. We might want to be more targeted? there is overhead to interning

Reply:
+1, I don't think we need weakIntern here.

Reply:
+1, we should consider can be repeatedly accessed and used